unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/AvatarAttach/Handler/AvatarAttachComponentHandler.cs

```diff
@@ -0,0 +1,169 @@
+﻿using System;
+using DCL.Components;
+using DCL.Configuration;
+using DCL.Controllers;
+using DCL.ECSRuntime;
+using DCL.Models;
+using DCL.Helpers;
+using UnityEngine;
+
+namespace DCL.ECSComponents
+{
+    public class AvatarAttachComponentHandler : IECSComponentHandler<PBAvatarAttach>
+    {
+        private const float BOUNDARIES_CHECK_INTERVAL = 5;
+        
+        internal PBAvatarAttach prevModel = null;
+
+        internal IAvatarAnchorPoints anchorPoints;
+        internal AvatarAnchorPointIds anchorPointId;
+        private IDCLEntity entity;
+        private IParcelScene scene;
+
+        private Action componentUpdate = null;
+
+        internal readonly GetAnchorPointsHandler getAnchorPointsHandler;
+        private readonly IUpdateEventHandler updateEventHandler;
+        internal ISceneBoundsChecker sceneBoundsChecker;
+        
+        private Vector2Int? currentCoords = null;
+        private bool isInsideScene = true;
+        private float lastBoundariesCheckTime = 0;
+
+        public AvatarAttachComponentHandler(IUpdateEventHandler updateEventHandler, ISceneBoundsChecker sceneBoundsChecker)
+        {
+            this.updateEventHandler = updateEventHandler;
+            this.sceneBoundsChecker = sceneBoundsChecker;
+            getAnchorPointsHandler = new GetAnchorPointsHandler();
+            getAnchorPointsHandler.OnAvatarRemoved += Detach;
+        }
+
+        public void OnComponentCreated(IParcelScene scene, IDCLEntity entity)
+        {
+            this.scene = scene;
+            this.entity = entity;
+        }
+
+        public void OnComponentRemoved(IParcelScene scene, IDCLEntity entity)
+        {
+            Dispose();
+        }
+
+        public void OnComponentModelUpdated(IParcelScene scene, IDCLEntity entity, PBAvatarAttach model)
+        {
+            // If is the same avatar, we skip
+            if (model == null || (prevModel != null && prevModel.AvatarId == model.AvatarId))
+                return;
```

Comment:
what would happen here if `AvatarId` is the same as before but `AnchorPointId` changed?

Reply:
This is a bug that is already happening in prod in the old component, I will fix it here as well